code-challenges.test.js

```diff
@@ -15,31 +15,63 @@
 
 // a) Create a test with an expect statement using the variable provided. HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-var colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+// Pseudo code:
+// Create a function takes an array and remove the first item from the array and shuffles remaining content
+// Use the following mutators:
+//    .shift - to remove the zeroth index ot the array
+//    .shuffle() method - to shuffle array
 
 
+describe("hue", () => {
+    var colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+
+    it("removes the first item from the array and shuffles the remaining content", () => {
+        expect(hue(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]));
+
+        expect(hue(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre",]));
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+const hue = (arr) => {
+    arr.shift()
+    return arr.sort((a, b) => 0.5 - Math.random())
+}
 
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
+// var nums1 = [3, 56, 90, -8, 0, 23, 6]
+// // Expected output: [-8, 90]
+// var nums2 = [109, 5, 9, 67, 8, 24]
+// // Expected output: [5, 109]
 
 
 
-// b) Create the function that makes the test pass.
+describe("cat", () => {
+    var nums1 = [3, 56, 90, -8, 0, 23, 6]
+    var nums2 = [109, 5, 9, 67, 8, 24]
+
+    it("takes an array of numbers.", () => {
+        expect(cat(nums1)).toEqual([-8, 90]);
+        expect(cat(nums2)).toEqual([5, 109]);
+    })
+})
+
+// Pseudo code:
+// create an array that takes in numbers
 
 
+// b) Create the function that makes the test pass.
+const cat = (arr) => {
+    return (sum(...numbers))
+}
+// This fuction is returing an error, not sure what I am doing wrong. Please advise.
+
```

Comment:
I don't think that `sum` is a function in JavaScript. (At least not to my knowledge.) And the goal is to find the largest value and the smallest value in a given array. So you would need to add up all the numbers you would need to find a way to evaluate them.